packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -597,30 +598,87 @@ function isDropDownChoice(item: QuestionnaireItem): boolean {
   );
 }
 
+/**
+ * This method is not called `areSameType` because it only checks if the two value's types are not explicitly different and allows one value to be `undefined`.
+ * @param value1 A value or `undefined`.
+ * @param value2 Another value to compare `value1`'s type to.
+ * @returns If the two values are NOT different types. It returns `true` if both are the same type or if `value1` is `undefined`.
+ */
+function areNotDifferentTypes(value1: TypedValue | undefined, value2: TypedValue): boolean {
+  // We return true for `undefined` since we handle the undefined case inside the case blocks
+  if (value1 === undefined || value1.type === value2.type) {
```

Comment:
```suggestion
  if (!value1 || value1.type === value2.type) {
```

Reply:
I think a strict comparison with `undefined` is the most correct here since we know that it will either be exactly `undefined` or a `TypedValue`

It should also be slightly faster since there is only one explicit comparison instead of the fuzzy compare via falsiness check with `!`